libopenage/rng/rng.cpp

```diff
@@ -48,58 +49,93 @@ void RNG::seed(uint64_t val) {
 
 
 /**
- * murmurhash3 bit avalancher - output is uniformly distributed
- * and changing one bit of input results in ~1/2 of output bits changed
- * This function is a good way of mixing up an input,
- * or mixing together two results with hash64(val1 ^ val2)
- *
- * https://en.wikipedia.org/wiki/MurmurHash
- */
-static uint64_t hash64(uint64_t val) {
-	// The two magic numbers have been determined heuristically.
-	// They avalanche all bits in val to within 0.25% bias.
-
-	val ^= val >> 33;
-	val *= 0xff51afd7ed558ccd;
-	val ^= val >> 33;
-	val *= 0xc4ceb9fe1a85ec53;
-	return val ^ (val >> 33);
+* Rotate left no carry
+*
+* http://blog.regehr.org/archives/1063
+*/
+inline uint64_t rotl64(uint64_t x, uint64_t n)
+{
+	assert (n<64);
+	return (x<<n) | (x>>(-n&63));
 }
 
-
-/*
- * reinterpret_casts 8 chars from a buffer as a uint64_t.
- */
-static inline uint64_t from_cptr(const char *data) {
-	uint64_t result;
-	memcpy(reinterpret_cast<char *>(&result), data, sizeof(result));
-	return result;
+// Maybe this should be a lambda inside of siphash()?
+inline void sipround(uint64_t *v) {
```

Comment:
I tried both options and the generated rng.cpp.o was the same apart of some register names. So I made it a lambda because of the scope, so now sipround() it's inside of siphash().